patches/minecraft/net/minecraft/inventory/IInventory.java.patch

```diff
@@ -0,0 +1,15 @@
+--- a/net/minecraft/inventory/IInventory.java
++++ b/net/minecraft/inventory/IInventory.java
+@@ -44,4 +44,12 @@
+    default int func_174922_i() {
+       return 0;
+    }
++
++   /**
++    * Do not create a new collection in this method.
++    * @return The types of recipes supported by this inventory.
++    */
++   default java.util.Collection<net.minecraftforge.common.crafting.RecipeType<?>> getRecipeTypes() {
```

Comment:
Can we handle this in a map in recipe manager instead, so both recipes and inventories can speciy the mapping relations.

Reply:
Inventories are no longer going to be able to specify recipe type.  The inventory will instead request those recipes it wants to match with.